### PR TITLE
Discord Signup: specs, excluding #launch

### PR DESCRIPTION
**Pull request by rshipp**
### Description

Discord Signup specs, excluding controller #launch action.

### Motivation and Context

Task: https://app.asana.com/0/0/1201263861485595/f

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by elenabaurkot — COMMENTED** [review]
lgtm 👍

File: spec/controllers/discord_signups_controller_spec.rb [hunk 1]
```diff
@@ -3,12 +3,185 @@
...
+          expect(flash[:notice]).to match /successfully published/
+        end
+
+        it 'redirects to edit page' do
+          expect(response).to redirect_to(edit_course_path(course))
+        end
+
+        it 'calls the API correctly' do
+          # Hardcoding the path so that if someone changes it they're forced to see this comment
+          # and consider that it will break all previously published Discord Signups assignments.
```
Comment (sadleb) [inline spec/controllers/discord_signups_controller_spec.rb:144]: Solid.

File: spec/models/discord_server_spec.rb [hunk 2]
```diff
@@ -1,5 +1,68 @@
...
+  it { should validate_presence_of :discord_server_id }
+  it { should validate_presence_of :webhook_id }
+  it { should validate_presence_of :webhook_token }
+
+  describe 'validating uniqueness' do
+    before { create :discord_server }
+
+    it { should validate_uniqueness_of(:name) }
+    # Weird workaround for unique validations with numeric values, we have to
+    # add `.case_insensitive` to the matcher.
```
Comment (sadleb) [inline spec/models/discord_server_spec.rb:14]: Noted.

**Review by sadleb — APPROVED** [review]
lgtm, but defer to @elenabaurkot

**Pull request merged**